src/plugins/discover/public/application/main/hooks/use_adhoc_data_views.ts

```diff
@@ -66,64 +54,41 @@ export const useAdHocDataViews = ({
    * This is to prevent duplicate ids messing with our system
    */
   const updateAdHocDataViewId = useCallback(
-    async (dataViewToUpdate: DataView) => {
-      const newDataView = await dataViews.create({ ...dataViewToUpdate.toSpec(), id: undefined });
-
-      dataViews.clearInstanceCache(dataViewToUpdate.id);
-      setAdHocDataViewList((prev) =>
-        prev.filter((d) => d.id && dataViewToUpdate.id && d.id !== dataViewToUpdate.id)
-      );
+    async (prevDataView: DataView) => {
+      const newDataView = await dataViews.create({ ...prevDataView.toSpec(), id: uuid() });
+      dataViews.clearInstanceCache(prevDataView.id);
 
-      // update filters references
-      const uiActions = await getUiActions();
-      const trigger = uiActions.getTrigger(UPDATE_FILTER_REFERENCES_TRIGGER);
-      const action = uiActions.getAction(UPDATE_FILTER_REFERENCES_ACTION);
+      updateFiltersReferences(prevDataView, newDataView);
 
-      action?.execute({
-        trigger,
-        fromDataView: dataViewToUpdate.id,
-        toDataView: newDataView.id,
-        usedDataViews: [],
-      } as ActionExecutionContext);
+      stateContainer.actions.replaceAdHocDataViewWithId(prevDataView.id!, newDataView);
+      await stateContainer.replaceUrlAppState({ index: newDataView.id });
 
-      savedSearch.searchSource.setField('index', newDataView);
```

Comment:
I've checked and it's not necessary since `stateContainer.replaceUrlAppState` triggers a change of stat and when this is done, the removed code is executed

Reply:
So when the ad hoc data view changes id, because of an change of fields, then in the end this code is executed: https://github.com/elastic/kibana/blob/74f5058f99e58fdb12d9d1eb3f829ce3f812c4be/src/plugins/discover/public/application/main/hooks/use_discover_state.ts#L198
so the removed line is not necessary